homeassistant/components/epson/media_player.py

```diff
@@ -115,11 +126,7 @@ async def set_unique_id(self):
             return False
         uid = await self._projector.get_serial_number()
         if uid:
-            self.hass.config_entries.async_update_entry(self._entry, unique_id=uid)
-            self._unique_id = uid
-            self.hass.async_create_task(
-                self.hass.config_entries.async_reload(self._entry.entry_id)
-            )
+            self.hass.async_create_task(set_new_unique_id(self.hass, uid, self._entry))
```

Comment:
```suggestion
            async_set_new_unique_id((self.hass, uid, self._entry)
```

Reply:
We know the `entity_id` of the entity here, so we could just pass that directly, I think (`self.entity_id`).